src/aux-records/AIController.spec.ts

```diff
@@ -369,6 +369,111 @@ describe('AIController', () => {
             expect(chatInterface.chat).not.toBeCalled();
         });
 
+        it('should return success when allowedModels includes the model', async () => {
+            chatInterface.chat.mockReturnValueOnce(
+                Promise.resolve({
+                    choices: [
+                        {
+                            role: 'user',
+                            content: 'test',
+                            finishReason: 'stop',
+                        },
+                    ],
+                    totalTokens: 1,
+                })
+            );
+
+            const result = await controller.chat({
+                model: 'test-model1',
+                messages: [
+                    {
+                        role: 'user',
+                        content: 'test',
+                    },
+                ],
+                temperature: 0.5,
+                userId,
+                userSubscriptionTier,
+            });
+
+            expect(result).toEqual({
+                success: true,
+                choices: [
+                    {
+                        role: 'user',
+                        content: 'test',
+                        finishReason: 'stop',
+                    },
+                ],
+            });
+            expect(chatInterface.chat).toBeCalledWith({
+                model: 'test-model1',
+                messages: [
+                    {
+                        role: 'user',
+                        content: 'test',
+                    },
+                ],
+                temperature: 0.5,
+                userId: 'test-user',
+            });
+        });
+
+        it('should return not_authorized when allowedModels does not include the model', async () => {
+            controller = new AIController({
+                chat: {
+                    interfaces: {
+                        provider1: chatInterface,
+                    },
+                    options: {
+                        defaultModel: 'default-model',
+                        defaultModelProvider: 'provider1',
+                        allowedChatModels: [
+                            {
+                                provider: 'provider1',
+                                model: 'modelA',
+                            },
+                            {
+                                provider: 'provider1',
+                                model: 'modelB',
+                            },
+                        ],
+                        allowedChatSubscriptionTiers: ['test-tier'],
+                    },
+                },
+                generateSkybox: null,
+                images: null,
+                metrics: store,
+                config: store,
+                hume: null,
+                sloyd: null,
+                policies: null,
+                policyController: policies,
+                records: store,
+            });
```

Comment:
@TroyceGowdy Same with this test. You shouldn't have to create a new AIController, instead you need to setup a subscription config.